smart-contracts/contracts/mixins/MixinTransfer.sol

```diff
@@ -188,6 +188,40 @@ contract MixinTransfer is
     return fee.mul(transferFeeBasisPoints) / BASIS_POINTS_DEN;
   }
 
+  /**
+  * @notice Modify the expirationTimestamp of a key
+  * by a given amount.
+  * @param _owner The owner of the key to modify
+  * @param _deltaT The amount of time in seconds by which
+  * to modify the keys expirationTimestamp
+  * @param _addTime Choose whether to increase or decrease
+  * expirationTimestamp (false == decrease, true == increase)
+  * @dev Throws if owner does not have a valid key.
+  */
+  function _timeMachine(
+    address _owner,
+    uint256 _deltaT,
+    bool _addTime
+  ) internal
+    hasValidKey(_owner)
```

Comment:
What if the destination has an expired key?
For partial transfer we would want to be able to add time to that account.  But not based on their current expiration - it should use block.timestamp + deltaT instead.

Reply:
Yes, agreed. I could use the `getHasValidKey` modifier here to check, and if the key is expired then use block.timestamp + deltaT instead.